src/Common/Core/Impl/OS/Win32NativeEnvironmentBlock.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Common.Core.OS {
+    public sealed class Win32NativeEnvironmentBlock : SafeBuffer {
```

Comment:
I meant that you may want to use `SafeBuffer` to read/write values directly. If you just need a `SafeHGlobalHandle`, `SafeHandleZeroOrMinusOneIsInvalid` as a base class is enough:

```cs
public sealed class SafeHGlobalHandle : SafeHandleZeroOrMinusOneIsInvalid {
    public int Length { get; }

    private SafeHGlobalHandle(int length) : base(true) {
        Length = length;
    }

    protected override bool ReleaseHandle() {
        Marshal.FreeHGlobal(handle);
        return true;
    }

    public static SafeHGlobalHandle Create(byte[] bytes) {
        var handle = Create(bytes.Length);
        Marshal.Copy(bytes, 0, result.DangerousGetHandle(), bytes.Length);
        return handle;
    }

    private static SafeHGlobalHandle Create(int length) {
        var handle = new SafeHGlobalHandle(length);
 
        RuntimeHelpers.PrepareConstrainedRegions();
        try { 
        } finally {
            var ptr = Marshal.AllocHGlobal(length);
            handle.SetHandle(ptr);
        }
        return handle;
    }
}
```